src/main/java/me/jellysquid/mods/lithium/mixin/block/redstone_wire/RedstoneWireBlockMixin.java

```diff
@@ -0,0 +1,205 @@
+ *   in the context of the network it is a part of. This means each wire in a network
+ *   could check and update its power level over half a dozen times. This also leads to
+ *   way more shape and block updates than necessary.
+ * <br>
+ * - It emits copious amounts of duplicate and redundant shape and block updates. While
+ *   the recursive updates are largely to blame, even a single state change leads to 18
+ *   redundant block updates and up to 16 redundant shape updates.
+ * 
+ * <p>
+ * Unfortunately fixing either of these aspects can be detected in-game, even if it is
+ * through obscure mechanics. Removing redundant block updates can be detected with
+ * something as simple as a redstone wire on a trapdoor, while removing the recursive
+ * updates can be detected with locational setups that rely on a specific block update
+ * order.
+ * 
+ * <p>
+ * What we can optimize, however, are the power calculations. In vanilla, these are split
+ * into two parts:
+ * <br>
+ * - Power from non-wire components.
+ * <br>
+ * - Power from other redstone wires.
+ * <br>
+ * We can combine the two to reduce calls to World.getBlockState and BlockState.isSolidBlock
+ * as well as calls to BlockState.getWeakRedstonePower and BlockState.getStrongRedstonePower.
+ * We can avoid calling those last two methods on redstone wires altogether, since we know
+ * they should return 0.
+ * <br>
+ * These changes can lead to a mspt reduction of up to 30% on top of Lithium's other
+ * performance improvements.
+ * 
+ * @author Space Walker
+ */
+@Mixin(RedstoneWireBlock.class)
+public class RedstoneWireBlockMixin extends Block {
+    
+    private static final Direction[] DIRECTIONS = Direction.values();
+    private static final Direction[] DIRECTIONS_VIRTICAL = { Direction.DOWN, Direction.UP };
+    private static final Direction[] DIRECTIONS_HORIZONTAL = { Direction.WEST, Direction.EAST, Direction.NORTH, Direction.SOUTH };
+    
+    private static final int MIN = 0;            // smallest possible power value
+    private static final int MAX = 15;           // largest possible power value
+    private static final int MAX_WIRE = MAX - 1; // largest possible power a wire can receive from another wire
+    
+    public RedstoneWireBlockMixin(Settings settings) {
+        super(settings);
+    }
+    
+    @Inject(
+            method = "getReceivedRedstonePower",
+            cancellable = true,
+            at = @At(
+                    value = "HEAD"
+            )
+    )
+    private void getReceivedPowerFaster(World world, BlockPos pos, CallbackInfoReturnable<Integer> cir) {
+        cir.setReturnValue(this.getReceivedPower(world, pos));
+        cir.cancel();
+    }
+    
+    /**
+     * Calculate the redstone power a wire at the given location receives from the
+     * blocks around it.
+     */
+    private int getReceivedPower(World world, BlockPos pos) {
+        int power = MIN;
+        
+        for (Direction dir : DIRECTIONS_VIRTICAL) {
```

Comment:
Also according to someone it should be `DIRECTIONS_VERTICAL`
*just me relaying info :P*